polkadot/node/service/src/overseer.rs

```diff
@@ -310,17 +318,28 @@ where
 			rand::rngs::StdRng::from_entropy(),
 		))
 		.approval_distribution(ApprovalDistributionSubsystem::new(
-			Metrics::register(registry)?,
+			approval_voting_parallel_metrics.approval_distribution_metrics(),
 			approval_voting_config.slot_duration_millis,
 			Arc::new(RealAssignmentCriteria {}),
+			enable_approval_voting_parallel,
 		))
 		.approval_voting(ApprovalVotingSubsystem::with_config(
-			approval_voting_config,
+			approval_voting_config.clone(),
 			parachains_db.clone(),
 			keystore.clone(),
 			Box::new(sync_service.clone()),
-			Metrics::register(registry)?,
+			approval_voting_parallel_metrics.approval_voting_metrics(),
 			Arc::new(spawner.clone()),
+			enable_approval_voting_parallel,
+		))
+		.approval_voting_parallel(ApprovalVotingParallelSubsystem::with_config(
```

Comment:
Can't we use a dummy subsystem if `enable_approval_voting_parallel` is false ?

Reply:
I was also confused by this initially, but it seems we can't as it would change the return type dynamically, which you can't do in Rust

Reply:
We can do it the way we did with `validator_overseer_builder `and `collator_overseer_builder` see my last comment: https://github.com/paritytech/polkadot-sdk/pull/4849#issuecomment-2340441122, which I think is a bit better than this subsystem_enabled/disabled flags that seems to confuse everyone :D.

Reply:
Something like this: https://github.com/paritytech/polkadot-sdk/commit/f9139ae5c317d2463f3bcaa2fdbf6d9b2ca32ddf#diff-940ead02e0e1bb61e415124ffcfe32630b246b4e66787d3a4606f6c692927f7eR770, which I think is a bit better than the subsystem_enabled/subsystem_disabled flags.